src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -144,52 +215,79 @@ sealed trait LazyModuleImpLike extends RawModule
   val auto: AutoBundle
   protected[diplomacy] val dangles: Seq[Dangle]
 
-  // .module had better not be accessed while LazyModules are still being built!
-  require (!LazyModule.scope.isDefined, s"${wrapper.name}.module was constructed before LazyModule() was run on ${LazyModule.scope.get.name}")
+  /** [[wrapper.module]] had better not be accessed while LazyModules are still being built! */
+  require (LazyModule.scope.isEmpty, s"${wrapper.name}.module was constructed before LazyModule() was run on ${LazyModule.scope.get.name}")
 
+  /** set name from [[wrapper]]. */
   override def desiredName = wrapper.desiredName
   suggestName(wrapper.suggestedName)
 
-  implicit val p = wrapper.p
+  /** [[Parameters]] for diplomatic [[Module]]s. */
+  implicit val p: Parameters = wrapper.p
 
+  /** [[instantiate]] will be called when a instance of [[LazyModuleImp]] is created. */
   protected[diplomacy] def instantiate() = {
     val childDangles = wrapper.children.reverse.flatMap { c =>
       implicit val sourceInfo = c.info
+      /** calling [[c.module]] will push the real [[Module]] into [[chisel3.internal.Builder]],
+        * notice this place is not calling [[instantiate]], it just push the [[c.module]] to Builder.
+        * */
       val mod = Module(c.module)
+      /** ask each child to finish instantiate. */
       mod.finishInstantiate()
+      /** return dangles of each child. */
       mod.dangles
     }
+
+    /** ask each node in the [[LazyModule]] to call [[BaseNode.instantiate]].
+      * it will return a sequence of [[Dangle]] of these [[BaseNode]].
+      * */
     val nodeDangles = wrapper.nodes.reverse.flatMap(_.instantiate())
+    /** add all node and child dangle together. */
     val allDangles = nodeDangles ++ childDangles
+    /** internal nodes pairing, group [[allDangles]] which has same [[Dangle.source]]*/
     val pairing = SortedMap(allDangles.groupBy(_.source).toSeq:_*)
+    /** connect dangles can be paired. */
     val done = Set() ++ pairing.values.filter(_.size == 2).map { case Seq(a, b) =>
       require (a.flipped != b.flipped)
+      /** @todo quite strange. */
       if (a.flipped) { a.data <> b.data } else { b.data <> a.data }
```

Comment:
Hello, @jackkoenig I have a question that, I used to think `<>` doesn't need to judge `this` and `that`, was that a misunderstanding?

Reply:
`chisel3` changed the behavior of `<>`. This may be related to this (either before or after the chisel3 change).

Reply:
Got it. Than you!

Reply:
So do we need `if (a.flipped) { a.data <> b.data } else { b.data <> a.data }` this line in `chisel3`, since this sounds to be a legacy code.

Reply:
your PR shouldn't be changing the code...

Reply:
OK. I revoke all changing of the code.